src/soroban_rpc.ts

```diff
@@ -63,6 +63,24 @@ export namespace SorobanRpc {
     oldestLedger: number;
     oldestLedgerCloseTime: number;
 
+    // the fields below are set if status is SUCCESS
+    ledger?: number;
```

Comment:
Maybe this should be a union type with of `GetTransactionResponseSuccess | GetTransactionResponseError`?